RAGchain/utils/linker/dynamo_linker.py

```diff
@@ -102,15 +103,29 @@ def handle_error(err, table_name, operation):
 
     def get_json(self, ids: list[Union[UUID, str]]):
         str_ids = [str(find_id) for find_id in ids]
-        response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
-        return [response['Item']['db_origin'] for response in response_list]
+        data_list = []
+        for find_id in str_ids:
+            # Check if id exists in dynamo linker
+            """
+            response_list = [self.table.get_item(Key={'id': find_id}) for find_id in str_ids]
+            return [response['Item']['db_origin'] for response in response_list]
+            """
+            data = self.table.get_item(Key={'id': find_id})
+            if 'Item' not in data:
+                warnings.warn(f"ID {find_id} not found in DynamoLinker", NoIdWarning)
+            else:
+                db_origin = data['Item']['db_origin']
+                # Check if data exists in dynamo linker
+                if db_origin is None:
+                    warnings.warn(f"Data {find_id} not found in DynamoLinker", NoDataWarning)
+                    continue
```

Comment:
If you don't append None when db_origin is None, it will occurs serious problem.
You request 10 ids, and get just 9 db_origin. You can get just 9 passages.
And maybe it is hard to know what passage is missing.

Reply:
return None, or just raise Error